src/YourEasyRent/Services/TelegramCallbackHandler.cs

```diff
@@ -60,81 +76,103 @@ public async Task HandleUpdateAsync(Update update)
             var callbackId = update.CallbackQuery.Id;
             var userResponse = update.CallbackQuery?.Data;
             var userId = update.CallbackQuery!.From.Id;
-
             var chatId = GetChatIdOrDefalt(userId);
 
-            var buttonName = update.CallbackQuery.Data;
+            if (buttonName == "Back")
+            {
+                _userStateManager.GoBackToOneStep(status);
+                var lastMenu = _userStateManager.MethodBackOnOneStep().ToString();
+                var lastMenuHandler = _buttonHandlers[lastMenu];
+                await lastMenuHandler.SendMenuToTelegramHandle(chatId);
+                return;
+
+            }
 
-            if (buttonName == "BrandMenu" || buttonName == "CategoryMenu")
+            if (buttonName == "BrandMenu")
             {
-                var handler = _buttonHandlers[buttonName];// если выбирать какую-то конкретную категорию или бренд, но выдается ошибка, тк в словаре _buttonHandlers нет этих значений
+                _currentMenuStatus = MenuStatus.BrandMenu;
+                var handler = _buttonHandlers[buttonName];
                 await handler.SendMenuToTelegramHandle(chatId);
                 return;
             }
 
-            _logger.LogInformation($"4. Received a  button'{buttonName}' userID {userId} and user First Name {firstName} and chatID {chatId}   и callbackId {callbackId}, userResponse {userResponse} ");
-           // Console.WriteLine($"4. Received a  button'{buttonName}' userID {userId} and user First Name {firstName} and chatID {chatId}   и callbackId {callbackId}, userResponse {userResponse} ");
+            if (buttonName == "CategoryMenu")
+            {
+                _currentMenuStatus = MenuStatus.CategoryMenu;
+                var handler = _buttonHandlers[buttonName];
+                await handler.SendMenuToTelegramHandle(chatId);
+                return;
+            }
 
             var botState = _userResponsesToChat[userId];
-            bool result = false;
-          
-            //var result = botState.PropertiesAreFilled();
-            //if (result == true) 
-            //{
-            //    string brand = botState.Brand;
-            //    string category = botState.Category;
-            //    await _actionsHandler.GetFilteredProductsMessage(brand, category);
-            //    return;
-
-            //}
 
             if (buttonName.StartsWith("Brand_"))
             {
-                botState.ChatId = update.CallbackQuery!.From.Id; // или ChatID?
+                botState.ChatId = update.CallbackQuery!.From.Id;
                 botState.Brand = buttonName.Replace("Brand_", "");
-                var categoryMenuHandler = _buttonHandlers["CategoryMenu"];
-                await categoryMenuHandler.SendMenuToTelegramHandle(chatId);
-                botState.Status = BotState.MenuStatus.BrandChosen;
+                _currentMenuStatus = MenuStatus.BrandChosen;
```

Comment:
All this logic should go inside userSateManager, so in the handler it looks kind of like this
           ```
 var botButton = new BotButton(update); // It's a new wrapper around update.CallbackQuery
if (botButton.IsBrand)
{
	var nextMenuStep =_userStateManager.SetBrandChosen(botButton.Value);

	if (nextStep.IsMenu)
	{
		var menu = _menuHandlers[nextMenuStep]; // used to be _buttonHandlers
		await menu.SendToChat(chatId);
		return;
	}
	
	// else we can also check nextStep.IsFinalResult
	var resultsMessage = _finalResultsService.GetFilteredProducts(_userStateManager.Brand, userStateManager.Category); // works with repo and build a message with products
	await _botClient.SendTextMessageAsync(botButton.ChatId, resultsMessage);
	return;
}
```